src/Chainweb/Miner/Coordinator.hs

```diff
@@ -135,6 +136,14 @@ resetPrimed :: MinerId -> ChainId -> PrimedWork -> PrimedWork
 resetPrimed mid cid (PrimedWork pw) = PrimedWork
     $! HM.update (Just . HM.insert cid Nothing) mid pw
 
+removePrimed
+    :: TVar PrimedWork
+    -> MinerId
+    -> ChainId
+    -> IO ()
+removePrimed pwVar mid cid =
```

Comment:
Why is this top-level

Reply:
From an intermediate version and some testing. I'll move it back down.